tests/trainer/test_trainer.py

```diff
@@ -989,6 +990,40 @@ def test_bnb_compile(self):
             with self.assertRaises(ValueError):
                 _ = Trainer(tiny_model, args, train_dataset=train_dataset)  # noqa
 
+
+    @require_peft
+    @require_bitsandbytes
+    def test_peft_submodule(self):
+        from peft import LoraConfig, get_peft_model
+
+        # Simply tests if initializing a Trainer with a PEFT on a submodule will pass _is_peft_model check.
+        # Throws error when quantised but not recognised as a peft model.
+        tiny_model = AutoModel.from_pretrained(
+            "hf-internal-testing/tiny-random-IdeficsModel", load_in_4bit=True
+        )
+
+        peft_config = LoraConfig(
+            r=8,
+            lora_alpha=32,
+            target_modules=["q_proj", "k_proj", "v_proj"],
+            lora_dropout=0.05,
+            bias="none",
+        )
+        tiny_model.vision_model = get_peft_model(tiny_model.vision_model, peft_config)
+
+        x = torch.randint(0, 100, (128,))
+        train_dataset = RepeatDataset(x)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            args = TrainingArguments(
+                tmp_dir,
+                learning_rate=1e-9,
+                logging_steps=5,
+            )
+            with self.assertRaises(ValueError):
+                _ = Trainer(tiny_model, args, train_dataset=train_dataset)  # noqa
```

Comment:
Wait - I'm confused 😅 

My understanding of this PR was that it's meant to allow training on quantized models which have a PEFT submodule. 

This tests doesn't obviously quantize (I might be missing something here). Is it meant to fail if the model isn't quantized?

Reply:
It does quantise the model. See `load_in_4bit=True` on line 1002.

Reply:
Ah, sorry, I missed that. In this case, why is this error being thrown? Shouldn't this model now be trainable (it's quantized and has a peft submodule).